include/oneapi/dpl/internal/random_impl/philox_engine.h

```diff
@@ -0,0 +1,544 @@
+    get_odd_elm_array(_Array __inp, std::index_sequence<_Is...>)
+    {
+        return std::array<scalar_type, std::index_sequence<_Is...>::size()>{std::get<_Is * 2 + 1>(__inp)...};
+    }
+
+  public:
+    /* Types */
+    using result_type = _UIntType;
+    using scalar_type = internal::element_type_t<result_type>;
+
+    /* Engine characteristics */
+    static constexpr std::size_t word_size = _w;
+    static constexpr std::size_t word_count = _n;
+    static constexpr std::size_t round_count = _r;
+
+    static_assert(_n == 2 || _n == 4, "_n must be 2 or 4");
+    static_assert(sizeof...(_consts) == _n, "the amount of _consts must be equal to _n");
+    static_assert(_r > 0, "_r must be more than 0");
+    static_assert(_w > 0 && _w <= std::numeric_limits<scalar_type>::digits,
+                  "_w must satisfy 0 < _w < std::numeric_limits<_UIntType>::digits");
+    static_assert(std::numeric_limits<scalar_type>::digits <= 64,
+                  "size of the scalar _UIntType (in case of sycl::vec<T, N> the size of T) must be less than 64 bits");
+    static_assert(std::is_unsigned_v<scalar_type>, "_UIntType must be unsigned type or vector of unsigned types");
+
+    static constexpr std::array<scalar_type, __array_size> multipliers =
+                     get_even_elm_array(std::array{_consts...},
+                                        std::make_index_sequence<__array_size>{});
+    static constexpr std::array<scalar_type, __array_size> round_consts =
+                     get_odd_elm_array(std::array{_consts...},
+                                       std::make_index_sequence<__array_size>{});
+    static constexpr scalar_type
+    min()
+    {
+        return 0;
+    }
+
+    static constexpr scalar_type
+    max()
+    {
+        return (std::numeric_limits<scalar_type>::max() & in_mask) - 1;
+    }
+    static constexpr scalar_type default_seed = 20111115u;
+
+    /* Constructors */
+    philox_engine() : philox_engine(default_seed) {}
+
+    explicit philox_engine(scalar_type __seed) { seed(__seed); }
+    
+    /* Seeding function */
+    void
+    seed(scalar_type __seed = default_seed)
+    {
+        seed_internal({__seed & in_mask});
+    }
+
+    /* Set the state to arbitrary position */
+    void
+    set_counter(const std::array<scalar_type, word_count>& __counter)
+    {
+        for (std::size_t __i = 0; __i < word_count; __i++)
+        {
+            // all counters are set in reverse order
+            state_.X[word_count - __i - 1] = __counter[__i] & in_mask;
+        }
+    }
+
+    /* Generating functions */
+    result_type
+    operator()()
+    {
+        result_type __ret = generate_internal<internal::type_traits_t<result_type>::num_elems>();
+        return __ret;
+    }
+    /* operator () overload for result portion generation */
+    result_type
+    operator()(unsigned int __random_nums)
+    {
+        result_type __ret = generate_internal<internal::type_traits_t<result_type>::num_elems>(__random_nums);
```

Comment:
same as above, can it be done in one line?

Reply:
Yep, done